stablehlo/tests/vhlo/stablehlo_legalize_to_vhlo.mlir

```diff
@@ -178,6 +178,18 @@ func.func @attr_dict() attributes {stablehlo.attr = {attr1 = 1 : i32, attr2 = 2
   return
 }
 
+// CHECK-LABEL: "attr_custom_call_api_version_typed_ffi"
+// CHECK-NEXT: (%[[ARG0:.*]]: {{.*}})
+func.func @attr_custom_call_api_version_typed_ffi(%arg0: tensor<f32>) -> tensor<f32> {
+  %0 = "stablehlo.custom_call"(%arg0) {
+    call_target_name = "foo",
+    backend_config= {bar = 42 : i32},
+    // CHECK: api_version = #vhlo<api_version_v1 API_VERSION_TYPED_FFI>
+    api_version = 4 : i32
+  } : (tensor<f32>) -> tensor<f32>
+  func.return %0 : tensor<f32>
+}
```

Comment:
Can you also add a test for `attr_custom_call_api_version_typed_ffi_default` which specifies API v4 but doesn't specify `backend_config`? I believe this will fail, I think we're missing some code in StablehloLegalizeToVhlo.cpp for inserting a proper default value for `dictionary|string`


Likely requires:

- [ ] Modify [here](https://github.com/openxla/stablehlo/blob/8a78ebb89bc6c5608f6f27a7a9ce687b3578b79d/stablehlo/transforms/StablehloLegalizeToVhlo.cpp#L698) in StableHLO->VHLO to instead call get `getOrDefault` method you wrote for populating the value.
- [ ] Modify [here](https://github.com/openxla/stablehlo/blob/8a78ebb89bc6c5608f6f27a7a9ce687b3578b79d/stablehlo/transforms/VhloLegalizeToStablehlo.cpp#L739) in VHLO->StableHLO if the value is a default value (empty str or empty dict)

Reply:
I validated this during testing and and test passed without the `backend_config` for APIV4

Reply:
will update the PR with required StablehloLegalizeToVhlo.cpp changes

Reply:
updated  StableHLO->VHLO and VHLO->StableHLO change. 
added APIv4 test without the backend_config.